packages/picasso.js/src/core/component/component-factory.js

```diff
@@ -145,6 +147,13 @@ function setUpEmitter(ctx, emitter, settings) {
   ctx.emit = (name, ...event) => emitter.emit(name, ...event);
 }
 
+function tearDownEmitter(ctx, emitter) {
+  (ctx.eventListeners || []).forEach(({ event, listener }) => {
+    emitter.removeListener(event, listener);
+    ctx.emit = () => {};
```

Comment:
No need to run this inside the forEach. There is only one ctx.

Reply:
fixed!